src/abstractdataframe/abstractdataframe.jl

```diff
@@ -587,13 +588,12 @@ function completecases(df::AbstractDataFrame, col::Union{Integer, Symbol})
     res
 end
 
-completecases(df::AbstractDataFrame, cols::Union{AbstractVector, Regex}) =
-    completecases(df[cols])
+completecases(df::AbstractDataFrame, cols::Union{AbstractVector, Regex, Not}) =
+    completecases(select(df, cols, copycols=false))
```

Comment:
Shouldn't  `df[cols]` support the same features as `select`?

Reply:
`df[cols]` and `select` have two differences:
* `select` allows us to avoid copying of columns and `df[cols]` always copies columns if `cols` is a collection (as in this case)
* `select` always returns an `AbstractDataFrame` and `df[cols]` can return a vector (this is not the case here as if a single column is passed we have a special faster method above)

In general these differences are the reasons why `select` exists at all, otherwise, we could use `getindex`.